### ISSUE TITLE
False positive for `AWS::ElasticLoadBalancingV2::Listener` `CertificateArn` when certificate is in IAM

### ISSUE THREAD

**Issue by pawelz99** [aws-cloudformation/cfn-lint#4551]
### CloudFormation Lint Version

1.51.5

### What operating system are you using?

Linux, MacOs

### Describe the bug

`cfn-lint` version 1.51.5 expects that `Properties.Certificates[].CertificateArn` in `AWS::ElasticLoadBalancingV2::Listener` matches ARN of certificate in Amazon Certificate Manager.
However, certificates can be stored in IAM too, see IAM documentation [Manage server certificates in IAM](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_server-certs.html).
CloudFormation template with following `AWS::ElasticLoadBalancingV2::Listener` passes CloudFormation template verification and stack can be successfully created _(account and certificate name redacted)_:
```yaml
 albListenerPort8443:
  Type: AWS::ElasticLoadBalancingV2::Listener
  Properties:
   Certificates:
   - CertificateArn: arn:aws:iam::000000000000:server-certificate/certificate_name_here
   SslPolicy: ELBSecurityPolicy-TLS13-1-2-Res-PQ-2025-09
   DefaultActions:
   - Type: fixed-response
     FixedResponseConfig:
      StatusCode: 404
   LoadBalancerArn: !Ref alb
   Port: 8443
   Protocol: HTTPS
```
`cfn-lint` reports error `E1159`:
```
E1159 'arn:aws:iam::000000000000:server-certificate/certificate_name_here' is not a 'AWS::ACM::Certificate.Arn' with pattern '^arn:aws[a-zA-Z-]*:acm:[a-z0-9-]+:\\d{12}:certificate/.+$'
```

AWS Elastic Load Balancing documentation [SSL certificates for your Application Load Balancer](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/https-listener-certificates.html#sni-certificate-list) does not mandate certificate has to be stored in ACM, while [Replace the default certificate](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/listener-update-certificates.html#replace-default-certificate) guide explicitly states:
> Within the ACM and IAM certificates table, select a new default certificate.

CloudFormation [AWS::ElasticLoadBalancingV2::Listener Certificate CertificateArn](https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/aws-properties-elasticloadbalancingv2-listener-certificate.html#cfn-elasticloadbalancingv2-listener-certificate-certificatearn) does not specify pattern limiting ARN to ACM certificates only.

### Expected behavior

`cfn-lint` allows `CertificateArn` ARNs from:
- ACM: `arn:aws:acm:region:account-id:certificate/certificate-id`
- IAM : `arn:aws:iam::account:server-certificate/certificate-name-with-path`

### Reproduction template

```yaml
---
AWSTemplateFormatVersion: 2010-09-09
Description: CloudFormation template
Resources:
 alb:
  Type: AWS::ElasticLoadBalancingV2::LoadBalancer
  Properties:
   Name: alb
   SecurityGroups:
   - sg-00000000000000000
   Scheme: internal
   Subnets:
   - subnet-00000000
   LoadBalancerAttributes:
   - Key: access_logs.s3.enabled
     Value: true
   - Key: access_logs.s3.bucket
     Value: !Sub bucket_name-${AWS::Region}
   - Key: idle_timeout.timeout_seconds
     Value: 60
   Tags:
   - Key: keep-idle-alb
     Value: true

 albListenerPort443:
  Type: AWS::ElasticLoadBalancingV2::Listener
  Properties:
   Certificates:
   - CertificateArn: arn:aws:iam::000000000000:server-certificate/certificate-name
   SslPolicy: ELBSecurityPolicy-TLS13-1-2-Res-PQ-2025-09
   DefaultActions:
   - Type: fixed-response
     FixedResponseConfig:
      StatusCode: 404
   LoadBalancerArn: !Ref alb
   Port: 443
   Protocol: HTTPS
```

**Comment by VishalGawade1** [comment]
I'd like to take this one if thats okay

**Comment by kddejong** [comment]
Thank you for reporting this. You're correct — `CertificateArn` on `AWS::ElasticLoadBalancingV2::Listener` accepts both ACM and IAM server certificate ARNs. This has been fixed in the schema source and will be included in the next release.

**Comment by kddejong** [comment]
Fixed https://github.com/aws-cloudformation/resource-provider-enhanced-schemas/pull/14
